src/api.rs

```diff
@@ -0,0 +1,152 @@
+use std::path::PathBuf;
+
+use crate::{
+    io::{
+        parser::{all_input_consumed, RuleParser},
+        OutputFileManager,
+    },
+    logical::{
+        execution::{
+            selection_strategy::{
+                dependency_graph::graph_positive::GraphConstructorPositive,
+                strategy_graph::StrategyDependencyGraph, strategy_round_robin::StrategyRoundRobin,
+            },
+            ExecutionEngine,
+        },
+        model::Identifier,
+    },
+    physical::management::database::TableId,
+};
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    /// nemo error
+    #[error(transparent)]
+    Nemo(#[from] crate::error::Error),
+    /// API error
+    #[error("API error {0} occurred: {1}")]
+    API(usize, String),
```

Comment:
Just no. You're re-inventing an `enum`, badly, with two different fields that can even drift separately. All of these should rather be variants in an enum, with the error message associated to the variant, and, should we really need an error code at all (what for? this is a rust API, after all, callers will be doing `match`es on the variants, and foreign language bindings are explicitly not in scope for this API), this can be done as a method on the error enum.